toolchain/sem_ir/typed_insts.h

```diff
@@ -558,6 +558,21 @@ struct ConvertToValueAction {
   TypeInstId target_type_inst_id;
 };
 
+// A type for C++ `void`. Should only be used for pointers (`void*`).
+struct CppVoidType {
+  static constexpr auto Kind = InstKind::CppVoidType.Define<Parse::NoneNodeId>(
+      {.ir_name = "cpp_void_type",
+       .is_type = InstIsType::Always,
+       .constant_kind = InstConstantKind::Always,
```

Comment:
Looking at this again, this should probably be `Conditional`. Because you're setting `Always`, the system assumes it's inherently complete, but that's not correct -- it should probably instead be treated as *never* complete.